tests/chainer_tests/functions_tests/rnn_tests/test_function_n_step_gru.py

```diff
@@ -1,91 +1,121 @@
-import unittest
-
 import numpy
 
 import chainer
-from chainer.backends import cuda
-from chainer import functions
-from chainer import gradient_check
+import chainer.functions as F
 from chainer import testing
-from chainer.testing import attr
+from chainer.testing import backend
 
 
 def sigmoid(x):
     return numpy.tanh(x * 0.5) * 0.5 + 0.5
 
 
-def _split(inputs, pos):
-    return inputs[:pos], inputs[pos:]
-
-
-def _to_gpu(x):
-    if x is None:
-        return None
-    elif isinstance(x, list):
-        return [_to_gpu(xi) for xi in x]
-    else:
-        return cuda.to_gpu(x)
-
-
-def _wrap_variable(x):
-    if isinstance(x, list):
-        return [_wrap_variable(xi) for xi in x]
-    else:
-        return chainer.Variable(x)
-
-
-class TestNStepGRU(unittest.TestCase):
-
-    batches = [3, 2, 1]
-    length = len(batches)
-    in_size = 3
-    out_size = 2
-    n_layers = 2
-    dropout = 0.0
+def array(shape, dtype):
+    return numpy.random.uniform(-1, 1, shape).astype(dtype)
+
+
+@testing.parameterize(*testing.product_dict(
+    [
+        {'n_layers': 1, 'hidden_size': 2,
+            'input_size': 1, 'batches': (1, 1, 1)},
+        {'n_layers': 2, 'hidden_size': 2,
+            'input_size': 3, 'batches': (3, 2, 1)},
+        {'n_layers': 4, 'hidden_size': 6,
+            'input_size': 3, 'batches': (5, 3, 1)},
+        {'n_layers': 5, 'hidden_size': 10,
+            'input_size': 6, 'batches': (6, 5, 3)},
+    ]))
+@testing.fix_random()
+@backend.inject_backend_tests(
+    None,
+    # ChainerX tests
+    testing.product({
+        'use_chainerx': [True],
+        'chainerx_device': ['native:0', 'cuda:0'],
+    })
+    # CPU tests
+    + testing.product({
+        'use_cuda': [False],
+        'use_ideep': ['never', 'always'],
+    })
+    # GPU tests
+    + testing.product([
+        [{'use_cuda': True}],
+
+        testing.product({
+            'use_cudnn': ['always', 'never'],
```

Comment:
```suggestion
        testing.product({
            'use_cudnn': ['never']})
        + testing.product({
            'use_cudnn': ['always'],
```

Otherwise you are increasing the test cases